### PR TITLE
Refactor credentials from error handling 

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #

## Proposed Changes
-  Remove all issues related non instantiated definition. https://github.com/Automattic/wp-calypso/pull/94682/files#diff-ced52ec24ed2cc9e31a92a4d7d40e71b1062061fe0abccc5b0c082418b372731L62
- Remove field name conflicts with the backend expectations. It was causing overcomplicated mappings between both definitions. ([code](https://github.com/Automattic/wp-calypso/pull/94682/files#diff-ced52ec24ed2cc9e31a92a4d7d40e71b1062061fe0abccc5b0c082418b372731L24-L45))
- Fix multiple rerenders issues caused by sequencias setErrors calls.   ([code](https://github.com/Automattic/wp-calypso/pull/94682/files#diff-ced52ec24ed2cc9e31a92a4d7d40e71b1062061fe0abccc5b0c082418b372731L70))
- Move error handling to a separate hook

## Testing Instructions
**Base:**
* Access any flow using the credential step (E.g site-migration/site-migration-credentials?from=[FROM]&siteSlug=[WP_COM_SITESLUG]&siteId=[WPCOM_SITE_ID]
* Follow the above scenarios

Scenario 1: (Credentials - Required Fields)
* Click on Continue without fill any field
* Check if you can see errors on the required fields (User and password)

Scenario 2: (Credentials - Create)
* Fill all fields with correct values
* Click on Continue
* Check if you are redirected to the next screen.

Scenario 3: (Backup file - Required Fields)
* Click on Continue without fill any field
* Check if you can see errors on the required fields (User and password)

Scenario 4: (Backup - Create)
* Fill all fields with correct values
* Click on Continue
* Check if you are redirected to the next screen

NOTE: These and more scenarios are tested via this test [suite](https://github.com/Automattic/wp-calypso/blob/82923614899b13bdba8ae9bfe109de03c20639a6/client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/test/index.tsx)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-credentials-form.ts [hunk 1]
```diff
@@ -1,124 +1,66 @@
...
 import { useEffect } from 'react';
 import { useForm } from 'react-hook-form';
 import { useQuery } from 'calypso/landing/stepper/hooks/use-query';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
-import { MigrationError, CredentialsFormData } from './types';
+import { useFormErrorMapping } from './hooks/use-form-error-mapping';
+import { CredentialsFormData } from './types';
 import { useSiteMigrationCredentialsMutation } from './use-site-migration-credentials-mutation';
 
-const mapApiError = ( error: any ) => {
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-credentials-form.ts:9]: I removed this response transformation to reduce complexity

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-credentials-form.ts [hunk 2]
```diff
@@ -1,124 +1,66 @@
...
-		onSuccess: () => {
-			recordTracksEvent( 'calypso_site_migration_automated_request_success' );
-			onSubmit();
-		},
-		onError: ( error: any ) => {
-			handleMigrationError( mapApiError( error ) );
-			recordTracksEvent( 'calypso_site_migration_automated_request_error' );
-		},
-	} );
+	const serverSideError = useFormErrorMapping( error, variables );
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-credentials-form.ts:19]: All error logic is now parsed by this hook and returned on the serverSideError variable

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-credentials-form.ts [hunk 3]
```diff
@@ -1,124 +1,66 @@
...
 		handleSubmit,
 		watch,
-		setError,
 		clearErrors,
 	} = useForm< CredentialsFormData >( {
 		mode: 'onSubmit',
 		reValidateMode: 'onSubmit',
 		disabled: isPending,
 		defaultValues: {
-			siteAddress: importSiteQueryParam,
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-credentials-form.ts:104]: The backend is waiting for a `from_url,` and the front is calling as `siteAddress.` It was causing constantly process of map values to show errors. I just removed it.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/components/site-address-field.tsx [hunk 4]
```diff
@@ -35,18 +35,18 @@ export const SiteAddressField: React.FC< Props > = ( {
 
 	return (
 		<div className="site-migration-credentials__form-field">
-			<FormLabel htmlFor="site-address">{ translate( 'Site address' ) }</FormLabel>
+			<FormLabel htmlFor="from_url">{ translate( 'Site address' ) }</FormLabel>
 			<Controller
 				control={ control }
-				name="siteAddress"
+				name="from_url"
```
Comment (Imran92) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/components/site-address-field.tsx:42]: We are switching to snake case here, but `backupFileLocation` is in camel case. Having two different cases for the form fields in the same form looks a bit confusing to me. WDYT about sticking to one of the cases for all form fields instead?

**Review by Imran92 — APPROVED** [review]
LGTM. Fixed some merge conflicts and some test failing on and off. Merging to unblock another ticket

**Pull request merged**